sdk/storage/azure-storage-queue/azure/storage/queue/models.py

```diff
@@ -298,44 +295,40 @@ class QueuePropertiesPaged(Paged):
         begin with the specified prefix.
     :param int results_per_page: The maximum number of queue names to retrieve per
         call.
-    :param str marker: An opaque continuation token.
+    :param str continuation_token: An opaque continuation token.
     """
-    def __init__(self, command, prefix=None, results_per_page=None, marker=None):
-        super(QueuePropertiesPaged, self).__init__(command, None)
+    def __init__(self, command, prefix=None, results_per_page=None, continuation_token=None):
+        super(QueuePropertiesPaged, self).__init__(
+            self._get_next_cb,
+            self._extract_data_cb,
+        )
+        self._command = command
         self.service_endpoint = None
         self.prefix = prefix
         self.current_marker = None
         self.results_per_page = results_per_page
-        self.next_marker = marker or ""
+        self.continuation_token = continuation_token or ""
         self.location_mode = None
 
-    def _advance_page(self):
-        """Force moving the cursor to the next azure call.
-
-        This method is for advanced usage, iterator protocol is prefered.
-
-        :raises: StopIteration if no further page
-        :return: The current page list
-        :rtype: list
-        """
-        if self.next_marker is None:
-            raise StopIteration("End of paging")
-        self._current_page_iter_index = 0
+    def _get_next_cb(self, continuation_token):
         try:
-            self.location_mode, self._response = self._get_next(
-                marker=self.next_marker or None,
+            return self._command(
+                marker=self.continuation_token or None,
```

Comment:
Why is the passed-in continuation token not used here?

Reply:
That's a good question, technically it is the same, because I didn't want to change too much your code, so I keep doing a subclass, on the same class that accept the callback (you see the callaback are `self._extract_data`).
For the sake of respecting the logic, you're right I should read the one in parameter.